From eb4e4b189a9a6ffab9e5bed55e73ee21d9bb28db Mon Sep 17 00:00:00 2001
From: Joone Hur <joone.hur@intel.com>
Date: Tue, 7 Oct 2014 13:31:07 -0700
Subject: [PATCH] Fix build warnings

../../third_party/webrtc/base/physicalsocketserver.cc:241:72: error: 'slevel' may be used uninitialized in this function [-Werror=maybe-uninitialized]
      int ret = ::getsockopt(s_, slevel, sopt, (SockOptArg)value, &optlen);
                                                                         ^
 ../../third_party/webrtc/base/physicalsocketserver.cc:241:72: error: 'sopt' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 ../../third_party/webrtc/base/physicalsocketserver.cc: In member function 'virtual int rtc::PhysicalSocket::SetOption(rtc::Socket::Option, int)':
 ../../third_party/webrtc/base/physicalsocketserver.cc:260:76: error: 'slevel' may be used uninitialized in this function [-Werror=maybe-uninitialized]
      return ::setsockopt(s_, slevel, sopt, (SockOptArg)&value, sizeof(value));
                                                                             ^
 ../../third_party/webrtc/base/physicalsocketserver.cc:260:76: error: 'sopt' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 cc1plus: all warnings being treated as errors
---
 base/physicalsocketserver.cc |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/base/physicalsocketserver.cc b/base/physicalsocketserver.cc
index cff5e4d..7008e73 100644
--- a/base/physicalsocketserver.cc
+++ b/base/physicalsocketserver.cc
@@ -233,8 +233,8 @@ class PhysicalSocket : public AsyncSocket, public sigslot::has_slots<> {
   }
 
   int GetOption(Option opt, int* value) {
-    int slevel;
-    int sopt;
+    int slevel = 0;
+    int sopt = 0;
     if (TranslateOption(opt, &slevel, &sopt) == -1)
       return -1;
     socklen_t optlen = sizeof(*value);
@@ -248,8 +248,8 @@ class PhysicalSocket : public AsyncSocket, public sigslot::has_slots<> {
   }
 
   int SetOption(Option opt, int value) {
-    int slevel;
-    int sopt;
+    int slevel = 0;
+    int sopt = 0;
     if (TranslateOption(opt, &slevel, &sopt) == -1)
       return -1;
     if (opt == OPT_DONTFRAGMENT) {
-- 
1.7.9.5

